common/faults/faults.c

```diff
@@ -0,0 +1,461 @@
+	fault_message_t recievedMessage = {msg_header_a->fault_sync_dashboard.latched, msg_header_a->fault_sync_dashboard.idx};
+	fault_message_t *currMessage = &messageArray[GET_IDX(recievedMessage.f_ID)];
+	handleCallbacks(recievedMessage, currMessage);
+}
+void fault_sync_precharge_CALLBACK(CanParsedData_t *msg_header_a) {
+	fault_message_t recievedMessage = {msg_header_a->fault_sync_precharge.latched, msg_header_a->fault_sync_precharge.idx};
+	fault_message_t *currMessage = &messageArray[GET_IDX(recievedMessage.f_ID)];
+	handleCallbacks(recievedMessage, currMessage);
+}
+void fault_sync_torque_vector_CALLBACK(CanParsedData_t *msg_header_a) {
+	fault_message_t recievedMessage = {msg_header_a->fault_sync_torque_vector.latched, msg_header_a->fault_sync_torque_vector.idx};
+	fault_message_t *currMessage = &messageArray[GET_IDX(recievedMessage.f_ID)];
+	handleCallbacks(recievedMessage, currMessage);
+}
+void fault_sync_l4_testing_CALLBACK(CanParsedData_t *msg_header_a) {
+	fault_message_t recievedMessage = {msg_header_a->fault_sync_l4_testing.latched, msg_header_a->fault_sync_l4_testing.idx};
+	fault_message_t *currMessage = &messageArray[GET_IDX(recievedMessage.f_ID)];
+	handleCallbacks(recievedMessage, currMessage);
+}
+//END AUTO RECIEVE FUNCTIONS
+
+//Force faults from daq
+void set_fault_CALLBACK(CanParsedData_t *msg_header_a) {
+    if (GET_OWNER(msg_header_a->set_fault.id) == currentMCU) {
+        forceFault(msg_header_a->set_fault.id, msg_header_a->set_fault.value);
+    }
+}
+
+//Return control back to this mcu from daq
+void return_fault_control_CALLBACK(CanParsedData_t *msg_header_a) {
+    if (GET_OWNER(msg_header_a->set_fault.id) == currentMCU) {
+        unForce(msg_header_a->set_fault.id);
+    }
+}
+
+
+//Updates faults owned by current mcu
+void updateFaults() {
+    uint16_t idx = ownedidx;
+    fault_attributes_t *fault;
+    do {
+        fault = &faultArray[idx];
+        //Fault is showing up as latched
+        if (((fault->tempLatch) && !(fault->message->latched))) {
+            //Has latching period ended
+            fault->message->latched = (fault->time_since_latch++ >= faultLatchTime[idx]) ? true : false;
+            if (fault->message->latched) {
+                currCount++;
+                switch(fault->priority) {
+                    case INFO:
+                        infoCount++;
+                        break;
+                    case WARNING:
+                        warnCount++;
+                        break;
+                    case CRITICAL:
+                        critCount++;
+                        break;
+                }
+                txFaultSpecific(idx);
+            }
+        }
+        //Fault is showing up as unlatched
+        else if (!(fault->tempLatch) && (fault->message->latched)) {
+            //Make sure unlatch period has elapsed
+            fault->message->latched = (fault->time_since_latch++ >= faultULatchTime[idx]) ? false : true;
```

Comment:
Do you mean to have the ++ after time_since_latch? It will increment the value after evaluating the expression

Reply:
That was how I intended it to be, as I wanted to evaluate the current value, then increase the fault latch time. I can change this if you'd like.